cmd/kaniko-acr/main.go

```diff
@@ -833,6 +903,55 @@ func handlePushOnly(c *cli.Context) error {
 	return nil
 }
 
+func setupAuthWithOIDC(tenantId, clientId, oidcIdToken, subscriptionId, registry, dockerUsername, dockerPassword, dockerRegistry string, noPush bool) (string, error) {
+	if registry == "" {
+		return "", fmt.Errorf("registry must be specified")
+	}
+	if tenantId == "" || clientId == "" || oidcIdToken == "" {
+		if noPush {
+			logrus.Warnf("NO_PUSH mode: insufficient OIDC params; tenant/client/oidc missing")
+			return "", nil
+		}
+		return "", fmt.Errorf("tenantId, clientId and OIDC token must be provided for OIDC authentication")
+	}
+
+	// Exchange OIDC ID token for AAD access token via client_assertion
+	aadToken, err := azureutil.GetAADAccessTokenViaClientAssertion(context.Background(), tenantId, clientId, oidcIdToken, "")
+	if err != nil {
+		if noPush {
+			logrus.Warnf("NO_PUSH mode: failed to get AAD token via OIDC: %v", err)
+			return "", nil
+		}
+		return "", errors.Wrap(err, "failed to get AAD token via OIDC")
+	}
+
+	// Best-effort public URL
+	publicUrl, err := getPublicUrl(aadToken, registry, subscriptionId)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to get public url with error: %s\n", err)
+	}
```

Comment:
public URL error is only logged to stderr, while in setupAuth it's handled differently